grant-project/grant-monorepo
# Functioning comments with EIP-712 signatures

Closes #182. Closes #6.

### What This Does

Implements everything needed for proposal comments:

* Correct EIP-712 format signature when you try to post a comment
* Comment creation endpoint that requires auth and validates EIP-712, along with matching text
* Handling of parent comment IDs for nested comments
* Hides comment form if logged out

Some other misc improvments:

* Domain checking for EIP-712.
* Improvements to the accuracy of the `User` type with the backend `UserSchema`

At some point I think the UI here is going to need a minor facelift, but it works for now.

### How to Test

Make sure you run `flask db upgrade`. Expects you to be on version `312db8611967`.

1. Go to a proposal logged out, confirm you can't comment
2. Login, return to proposal, and confirm you can comment
3. Create a new comment, confirm it showed up
4. Create a reply, confirm it showed up
5. Refresh the page, confirm the comments loaded in correctly

## Screenshots

### Comment EIP-712 request

<img width="472" alt="screen shot 2018-11-08 at 2 30 16 pm" src="https://user-images.githubusercontent.com/649992/48222887-b99ad680-e363-11e8-887f-14faf165037b.png">


### Nested comments

<img width="508" alt="screen shot 2018-11-08 at 2 16 04 pm" src="https://user-images.githubusercontent.com/649992/48222293-42187780-e362-11e8-94fc-773e143b3d48.png">
